prov/mrail/src/mrail_ep.c

```diff
@@ -47,11 +47,17 @@ static ssize_t mrail_recv(struct fid_ep *ep_fid, void *buf, size_t len,
 	struct mrail_ep *mrail_ep = container_of(ep_fid, struct mrail_ep,
 					     util_ep.ep_fid.fid);
 	uint32_t rail = mrail_get_rx_rail(mrail_ep);
+	void *rail_desc = NULL;
 	ssize_t ret;
 
 	assert(!src_addr);
 
-	ret = fi_recv(mrail_ep->eps[rail], buf, len, desc, 0, context);
+	if (desc) {
+		struct mrail_mr *mrail_mr = desc;
+		rail_desc = fi_mr_desc(mrail_mr->mrs[rail]);
+	}
```

Comment:
Move this into a static inline function or macro that either returns the underlying descriptor or null if none is needed.